sklearn/inspection/tests/test_permutation_importance.py

```diff
@@ -435,3 +436,29 @@ def my_scorer(estimator, X, y):
                                scoring=my_scorer,
                                n_repeats=1,
                                sample_weight=w)
+
+
+@pytest.mark.parametrize("input_scorers", [
+    ["r2", "neg_mean_squared_error"],
+    {
+        "r2": get_scorer("r2"),
+        "neg_mean_squared_error": get_scorer("neg_mean_squared_error")
+    }])
```

Comment:
I assume that the callable case could be the following:

```python
lambda estimator, X, y: {
        "r2": r2_score(y, estimator.predict(X)),
        "neg_mean_squared_error": -mean_absolute_error(y, estimator.predict(X))
    },
```
However, to check strictly the name of the function we would need to pass explicitly the name since `set(lambda)` would not work there :)

I see that testing this way is making the test fail. I will debug a bit more to see if I made a mistake or if we return another shape after computing the score.

Reply:
The issue is that we call `check_scoring` on callable while a callable returning a dict should make use of `_MultimetricScorer` maybe. I have to check how do we handle this case normally.